lib/core/src/lib/comments/comments.component.ts

```diff
@@ -46,141 +51,127 @@ export class CommentsComponent implements OnChanges {
 
-    ngOnChanges(changes: SimpleChanges) {
-        this.taskId = null;
-        this.nodeId = null;
+    ngOnChanges(changes: SimpleChanges): void {
+        this.id = null;
 
-        this.taskId = changes['taskId'] ? changes['taskId'].currentValue : null;
-        this.nodeId = changes['nodeId'] ? changes['nodeId'].currentValue : null;
+        this.id = changes['id'] ? changes['id'].currentValue : null;
 
-        if (this.taskId || this.nodeId) {
+        if (this.id) {
             this.getComments();
         } else {
             this.resetComments();
         }
     }
 
-    private getComments(): void {
+    getComments() {
         this.resetComments();
-        if (this.isATask()) {
-            this.commentProcessService.getTaskComments(this.taskId).subscribe(
-                (comments: CommentModel[]) => {
-                    if (comments && comments instanceof Array) {
-                        comments = comments.sort((comment1: CommentModel, comment2: CommentModel) => {
-                            const date1 = new Date(comment1.created);
-                            const date2 = new Date(comment2.created);
-                            return date1 > date2 ? -1 : date1 < date2 ? 1 : 0;
-                        });
-                        comments.forEach((currentComment) => {
-                            this.commentObserver.next(currentComment);
-                        });
-                    }
 
-                },
-                (err) => {
-                    this.error.emit(err);
+        if (!this.hasId()) {
+            return;
+        }
+
+        this.commentsService.get(this.id).subscribe(
+            (comments: CommentModel[]) => {
+                if (!this.isArrayInstance(comments)) {
+                    return;
                 }
-            );
+
+                comments = this.sortedComments(comments);
+                this.addCommentsToObserver(comments);
+
+            },
+            (err) => {
+                this.error.emit(err);
+            }
+        );
+    }
+
+    addComment() {
+        if (!this.canAddComment()) {
+            return;
         }
 
-        if (this.isANode()) {
-            this.commentContentService.getNodeComments(this.nodeId).subscribe(
-                (comments: CommentModel[]) => {
-                    if (comments && comments instanceof Array) {
-
-                        comments = comments.sort((comment1: CommentModel, comment2: CommentModel) => {
-                            const date1 = new Date(comment1.created);
-                            const date2 = new Date(comment2.created);
-                            return date1 > date2 ? -1 : date1 < date2 ? 1 : 0;
-                        });
-                        comments.forEach((comment) => {
-                            this.commentObserver.next(comment);
-                        });
-                    }
+        const comment: string = this.sanitize(this.message);
+
+        this.beingAdded = true;
```

Comment:
this one is not set back to false in case of successful result... maybe needs some unit testing as well

Reply:
Is set back to `false` in the finally block https://github.com/Alfresco/alfresco-ng2-components/blob/5c30ca84d273065a9b92b723f6991a09d2251e4a/lib/core/src/lib/comments/comments.component.ts#L128 but of course we'll add a unit test